apple/Sources/FerrostarMapLibreUI/Extensions/View.swift

```diff
@@ -0,0 +1,70 @@
+import SwiftUI
+
+extension View {
+    /// Given the view's `geometry`, synchronizes `childInsets`, such that
+    /// accumulating the child's insets with the parents insets, will be at least `minimumInset`.
+    ///
+    /// ```
+    ///    Given a minimumInset of 16:
+    ///    +-------------------------------------------------------------+
+    ///    |                       `parentGeometry`                      |
+    ///    |   +-----------------------------------------------------+   |
+    ///    |   |     `parentGeometry.safeAreaInsets` (Top: 16)       |   |
+    ///    |   |   +---------------------------------------------+   |   |
+    ///    |   |   |     insets added by this method (Top: 0)    |   |   |
+    ///    |   |   |   +------------------------------------+    |   |   |
+    ///    |   |   |   |                                    |    |   |   |
+    ///    |   | 8 | 8 |        child view (self)           | 16 | 0 |   |
+    ///    |   |   |   |                                    |    |   |   |
+    ///    |   |   |   +------------------------------------+    |   |   |
+    ///    |   |   |    insets added by this method (Bottom: 0)  |   |   |
+    ///    |   |   +---------------------------------------------+   |   |
+    ///    |   |    `parentGeometry.safeAreaInsets` (Bottom: 20)     |   |
+    ///    |   +-----------------------------------------------------+   |
+    ///    |                                                             |
+    ///    +-------------------------------------------------------------+
+    /// ```
+    func complementSafeAreaInsets(parentGeometry: GeometryProxy, minimumInset: CGFloat = 16) -> some View {
```

Comment:
Perhaps I'm over-engineering this, but my first question after reading the docs was which edge insets the number applied to. I think we can resolve that ambiguity AND make this more flexible by making it accept edge insets rather than a single number.